sortbystring.cpp

```diff
@@ -0,0 +1,38 @@
+#include <array>
+#include <iostream>
+#include <unordered_map>
+std::string sortByStrings(std::string s, std::string t) {
+  std::unordered_map<char, int> dic;
+  // Create an hash table of characters and how many of them there are in the string s.
+  for (int i = 0; i < s.size(); i++) {
+    auto itr = dic.find(s[i]);
+    if(itr == dic.end()){
+      dic.emplace(s[i], 1);
+    }
+    else{
+      itr->second++;
+    }
+
+  }
+  std::string to_return = "";
+  // Check if each charater in t is in s.
+  for(int i = 0; i < t.size(); i++){
+    auto itr = dic.find(t[i]);
+    if(itr != dic.end()){
+      // if a character from t is in s appened it to 'to_return'.
+      // there can be multiple copies of the strings in s so append them until itr->second becomes 0.
+      while(itr->second != 0){
+        to_return += itr->first;
+        itr->second--;
+      }
+    }
+  }
+  // return the string.
+  return to_return;
+}
+
+
+int main(int argc, char const *argv[]) {
+    std::cout << sortByStrings( "weather" ,"therapyw") << '\n';
+    return 0;
```

Comment:
`return 0` unnecessary in c++, falling off the end of the main function automatically returns zero.